apps/jobs/views.py

```
@@ -491,146 +492,21 @@ def leaderboard(request, challenge_phase_split_id):
-        "submission__participant_team__team_url",
-        "submission__is_baseline",
-        "challenge_phase_split",
-        "result",
-        "error",
-        "filtering_score",
-        "filtering_error",
-        "leaderboard__schema",
-        "submission__submitted_at",
-        "submission__method_name",
+    challenge_obj = challenge_phase_split.challenge_phase.challenge
+    distinct_sorted_leaderboard_data, http_status_code = calculate_distinct_sorted_leaderboard_data(
+        request.user,
+        challenge_obj,
+        challenge_phase_split,
+        only_public_entries=True,
     )
-
-    if challenge_phase_split.visibility == ChallengePhaseSplit.PUBLIC:
-        leaderboard_data = leaderboard_data.filter(submission__is_public=True)
-
-    all_banned_participant_team = []
-    for leaderboard_item in leaderboard_data:
-        participant_team_id = leaderboard_item["submission__participant_team"]
-        participant_team = ParticipantTeam.objects.get(id=participant_team_id)
-        all_participants_email_ids = (
-            participant_team.get_all_participants_email()
-        )
-        for participant_email in all_participants_email_ids:
-            if participant_email in all_banned_email_ids:
-                all_banned_participant_team.append(participant_team_id)
-                break
-        if leaderboard_item["error"] is None:
-            leaderboard_item.update(filtering_error=0)
-
-    if challenge_phase_split.show_leaderboard_by_latest_submission:
-        sorted_leaderboard_data = leaderboard_data
-    else:
-        sorted_leaderboard_data = sorted(
-            leaderboard_data,
-            key=lambda k: (
-                float(k["filtering_score"]),
-                float(-k["filtering_error"]),
-            ),
-            reverse=True
-            if challenge_phase_split.is_leaderboard_order_descending
-            else False,
+    if http_status_code == status.HTTP_400_BAD_REQUEST:
```

I think I am missing something here. If response is 400, why we want to return the data? :/

Also, can you please add a comment above this line to say what action to take when response is 400?

---

We want to return the error message. The variable name was wrong. I've changed it.